tcp collector crashes application

### Current Behavior

Enabling "tcp" collector results in windows_exporter crash in snapshot builds / master branch.

### Expected Behavior

Enabling "tcp" not crashing windows_exporter.

### Steps To Reproduce

```Markdown
Use snapshot build from master branch.
Enable "tcp" exporter.
Start windows_exporter.
```


### Environment

- windows_exporter Version: 0.29.2-35-gfbce64e
- Windows Server Version: Windows Server 2022 Datacenter


### windows_exporter logs

```shell
C:\Program Files\windows_exporter>windows_exporter_snapshot.exe --collectors.enabled "tcp" --log.level=debug --log.file=stdout --log.format=json
{"ts":"2024-10-15T11:44:41.774Z","level":"debug","caller":"exporter.go(main.run):174","msg":"Logging has Started"}
{"ts":"2024-10-15T11:44:41.774Z","level":"debug","caller":"exporter.go(main.setPriorityWindows):337","msg":"setting process priority to normal"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x12b0915]

goroutine 23 [running]:
github.com/prometheus-community/windows_exporter/internal/perfdata/v1.(*Collector).Collect(0xc0000a6da0)
        D:/a/windows_exporter/windows_exporter/internal/perfdata/v1/collector.go:51 +0x55
github.com/prometheus-community/windows_exporter/internal/perfdata/v1.NewCollector({0x160d8bb, 0x5}, {0x0?, 0x0?, 0x0?}, {0x0?, 0x0?, 0x0?})
        D:/a/windows_exporter/windows_exporter/internal/perfdata/v1/collector.go:30 +0x89
github.com/prometheus-community/windows_exporter/internal/perfdata.NewCollector(0x0?, {0x160d8bb?, 0x0?}, {0x0?, 0x0?, 0x0?}, {0xc0001dde88?, 0x0?, 0x0?})
        D:/a/windows_exporter/windows_exporter/internal/perfdata/perfdata.go:34 +0x98
github.com/prometheus-community/windows_exporter/internal/collector/tcp.(*Collector).Build(0xc00023a240, 0x0?, 0x0?)
        D:/a/windows_exporter/windows_exporter/internal/collector/tcp/tcp.go:118 +0x174
github.com/prometheus-community/windows_exporter/pkg/collector.(*MetricCollectors).Build.func1()
        D:/a/windows_exporter/windows_exporter/pkg/collector/collector.go:198 +0x8a
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*MetricCollectors).Build in goroutine 1
        D:/a/windows_exporter/windows_exporter/pkg/collector/collector.go:195 +0x225
```


### Anything else?

_No response_

--- a/internal/collector/tcp/tcp_test.go
+++ b/internal/collector/tcp/tcp_test.go
@@ -10,3 +10,7 @@ import (
 func BenchmarkCollector(b *testing.B) {
 	testutils.FuncBenchmarkCollector(b, tcp.Name, tcp.NewWithFlags)
 }
+
+func TestCollector(t *testing.T) {
+	testutils.TestCollector(t, tcp.New, nil)
+}

--- a/internal/perfdata/v1/collector.go
+++ b/internal/perfdata/v1/collector.go
@@ -44,7 +44,7 @@ func (c *Collector) Collect() (map[string]map[string]perftypes.CounterValues, er
 		return nil, fmt.Errorf("QueryPerformanceData: %w", err)
 	}
 
-	if len(perfObjects) == 0 {
+	if len(perfObjects) == 0 || perfObjects[0] == nil || len(perfObjects[0].Instances) == 0 {
 		return map[string]map[string]perftypes.CounterValues{}, nil
 	}
 

--- a/internal/perfdata/v1/perflib.go
+++ b/internal/perfdata/v1/perflib.go
@@ -418,6 +418,10 @@ func QueryPerformanceData(query string, counterName string) ([]*PerfObject, erro
 			}
 		}
 
+		if counterName != "" {
+			return objects[i : i+1], nil
+		}
+
 		// Next perfObjectType
 		objOffset += int64(obj.TotalByteLength)
 		numFilteredObjects++

